internal/setup/setup.go

```diff
@@ -418,14 +423,17 @@ func parseRelease(baseDir, filePath string, data []byte) (*Release, error) {
 		return nil, fmt.Errorf("%s: no archives defined", fileName)
 	}
 
-	keyringsByName := make(map[string]openpgp.KeyRing, 0)
-
-	for keyringName, keyringArmored := range yamlVar.Keyrings {
-		keyring, err := openpgp.ReadArmoredKeyRing(strings.NewReader(keyringArmored))
+	// Decode the public keys and match against provided IDs.
+	pubKeys := make(map[string]*packet.PublicKey, len(yamlVar.PublicKeys))
+	for keyName, yamlPubKey := range yamlVar.PublicKeys {
+		key, err := DecodePublicKey([]byte(yamlPubKey.Armor))
 		if err != nil {
-			return nil, fmt.Errorf("%s: cannot parse keyring %q: %w", fileName, keyringName, err)
+			return nil, fmt.Errorf("%s: invalid public key %q: %w", fileName, keyName, err)
 		}
-		keyringsByName[keyringName] = keyring
+		if yamlPubKey.KeyID != key.KeyIdString() {
+			return nil, fmt.Errorf("%s: invalid public key %q: key-id does not match", fileName, keyName)
```

Comment:
There's no key-id field. Also, maybe we can phrase it a bit more clearly:

`"public key %q armor has incorrect ID: expected %q, got %q"`

Reply:
Agree, this way we can show the errors directly without the user having to go back to the yaml